wagtail/images/models.py

```diff
@@ -1233,7 +1263,16 @@ def url(self):
 
     @property
     def alt(self):
-        return self.image.default_alt_text
+        # 'decorative' and 'contextual_alt_text' exist only for ImageBlock
+        if hasattr(self.image, "decorative") and self.image.decorative:
+            return ""
+        elif (
+            hasattr(self.image, "contextual_alt_text")
+            and self.image.contextual_alt_text
+        ):
+            return self.image.contextual_alt_text
+        else:
+            return self.image.default_alt_text
```

Comment:
I prefer the version with the explicit `decorative` flag, so that we're not reliant on distinguishing between the empty string case ("output blank alt text") and None / undefined ("fall back on the default alt text") - that seems an easy place to slip up. I'd also rather avoid reshuffling any logic at this late stage.